src/embind/embind.js

```diff
@@ -97,23 +97,93 @@ var LibraryEmbind = {
     throw new UnboundTypeError(message + ': ' + unboundTypes.map(getTypeName).join([', ']));
   },
 
+  $instancesCheck__deps: [],
+  $instancesCheck: function(field, object) {
+    var tempObject = object;
+    var isMatched = false;
+    while (true) {
+      if (tempObject.value !== undefined) { // Enum
+        if (tempObject.constructor.name.split('_')[0] === field) {
+          isMatched = true;
+        }
+        break;
+      } else if (
+        tempObject.constructor.name === field
+        || 'shared_ptr<' + tempObject.constructor.name + '>' === field
```

Comment:
This seems brittle. Is there some other way we can handle pointers? Maybe, once the instanceof works we can get a ref to the class that the pointer points to.

Reply:
I deleted instancesCheck function and now i use `instanceof` instead.